libsolidity/ast/Types.cpp

```diff
@@ -406,28 +603,52 @@ bool IntegerConstantType::isImplicitlyConvertibleTo(Type const& _convertTo) cons
 			return true;
 		return false;
 	}
+	else if (_convertTo.category() == Category::FixedPoint)
+	{
+		auto targetType = dynamic_cast<FixedPointType const*>(&_convertTo);
+		if (m_value == 0)
+			return true;
+		int forSignBit = (targetType->isSigned() ? 1 : 0);
+		if (m_value > 0)
+		{
+			bool properlyScaledBits = m_scalingFactor <= targetType->fractionalBits() ? 
+				true : m_scalingFactor == 1 && targetType->fractionalBits() == 0 ? true : false;
+			if (m_value <= (u256(-1) >> (256 - targetType->numBits() + forSignBit)) && properlyScaledBits)
+				return true;
+			else if (targetType->isSigned() && -m_value <= (u256(1) << (targetType->numBits() - forSignBit)) && properlyScaledBits)
+				return true;
+			return false;
+		}
+
+	}
 	else if (_convertTo.category() == Category::FixedBytes)
 	{
 		FixedBytesType const& fixedBytes = dynamic_cast<FixedBytesType const&>(_convertTo);
 		return fixedBytes.numBytes() * 8 >= integerType()->numBits();
 	}
-	else
-		return false;
+	return false;
 }
 
-bool IntegerConstantType::isExplicitlyConvertibleTo(Type const& _convertTo) const
+bool ConstantNumberType::isExplicitlyConvertibleTo(Type const& _convertTo) const
 {
-	TypePointer intType = integerType();
-	return intType && intType->isExplicitlyConvertibleTo(_convertTo);
+	if (m_value.denominator() == 1)
+	{
+		TypePointer intType = integerType();
+		return intType && intType->isExplicitlyConvertibleTo(_convertTo);
+	}
+	TypePointer fixType = fixedPointType();
```

Comment:
If only integer types are concerned, we disallowed conversion when the number was too large.
Now that we add rationals, we cannot disallow conversion for rationals that cannot be represented in finite precision.
So the conversion rules are a bit more complicated and we have to check each of them here.